src/engine/Match_tainting_mode.ml

```diff
@@ -193,18 +193,9 @@ let mk_specialized_formula_cache (rules : R.taint_rule list) =
   let count_tbl = Formula_tbl.create 128 in
   let flat_formulas =
     rules
-    |> List.concat_map (fun rule ->
+    |> List.concat_map (fun (rule : R.taint_mode R.rule_info) ->
            let (`Taint (spec : R.taint_spec)) = rule.R.mode in
-           List_.map (fun source -> source.R.source_formula) (snd spec.sources)
-           @ List_.map
-               (fun sanitizer -> sanitizer.R.sanitizer_formula)
-               (match spec.sanitizers with
-               | None -> []
-               | Some (_, sanitizers) -> sanitizers)
-           @ List_.map (fun sink -> sink.R.sink_formula) (snd spec.sinks)
-           @ List_.map
-               (fun propagator -> propagator.R.propagator_formula)
-               spec.propagators)
+           R.formula_of_mode (`Taint spec))
```

Comment:
Are these two statements different than `R.formula_of_mode rule.R.mode`?

Reply:
What two statements?

Reply:
Maybe I'm confused because of how GitHub is showing me things.

I'm referring to:

```
let (`Taint (spec : R.taint_spec)) = rule.R.mode in
R.formula_of_mode (`Taint spec))
```

vs just `R.formula_of_mode rule.R.mode`

Reply:
That's a coercion to please the type system. Take a look at the type `Rule.rule`, and `Rule.rule_info`. The `rule_info` type is parameterized by the kind of rules that are allowed. If you write what you propose, you get this type error:

```
This expression has type R.taint_mode but an expression was expected of type
  R.mode
The first variant type does not allow tag(s)
`Extract, `Search, `Secrets, `Steps
```

Reply:
Ah, got it.